lib/Check.vala

```diff
@@ -36,13 +35,45 @@ public abstract class ValaLint.Check {
     public abstract string get_description ();
 
     /**
-     * Checks a given line for formatting mistakes.
+     * Checks a given parse result for formatting mistakes.
      *
+     * @param parse_result The parsed string.
      * @param mistake_list The list new mistakes should be added to.
-     * @param line_index The index of the given line.
-     * @param line The line that should be checked.
+     */
+    public abstract void check (Gee.ArrayList<ParseResult?> parse_result, Gee.ArrayList<FormatMistake?> mistake_list);
+
+    /**
+     * Adds a mistake based on a regex pattern.
      *
-     * @return Indicates if new mistakes were added to the list.
+     * @param check The used check.
+     * @param pattern The regex pattern.
+     * @param mistake The mistake description.
+     * @param parse_result The current parse result element.
+     * @param mistakes The mistakes list.
+     * @param char_offset The offset between the mistake char position and the regex pattern start.
      */
-    public abstract bool check_line (Gee.ArrayList<FormatMistake?> mistake_list, int line_index, string line);
+    public static void add_regex_mistake (Check check, string pattern, string mistake, ParseResult parse_result, Gee.ArrayList<FormatMistake? > mistakes, int char_offset = 0) {
+
+        MatchInfo match_info;
+        try {
+            var regex = new Regex (pattern);
```

Comment:
Perhaps this regex could be made to be multiline to make matching things on a single line easier? This would make your trailing whitespace check simpler and shouldn't affect the others.